df_py/volume/calc_rewards.py

```diff
@@ -1,505 +1,54 @@
-            if nft_addr not in rewardsinfo[chainID]:
-                rewardsinfo[chainID][nft_addr] = {}
-            rewardsinfo[chainID][nft_addr][LP_addr] = R[i, j]
-
-    return rewardsperlp, rewardsinfo
-
-
-@enforce_types
-def _get_chain_nft_tups(
-    stakes: Dict[int, Dict[str, Dict[str, float]]],
-    nftvols_USD: Dict[int, Dict[str, str]],
-) -> List[Tuple[int, str]]:
-    """
-    @arguments
-      stakes - dict of [chainID][nft_addr][LP_addr] : veOCEAN_float
-      nftvols_USD -- dict of [chainID][nft_addr] : vol_USD_float
-
-    @return
-      chain_nft_tups -- list of (chainID, nft_addr), indexed by j
-    """
-    chainIDs = list(stakes.keys())
-    nft_addrs = _get_nft_addrs(nftvols_USD)
-    chain_nft_tups = [
-        (chainID, nft_addr)  # all (chain, nft) tups with stake
-        for chainID in chainIDs
-        for nft_addr in nft_addrs
-        if nft_addr in stakes[chainID]
-    ]
-    return chain_nft_tups
+from df_py.util.constants import DO_PUBREWARDS, DO_RANK
+from df_py.volume import allocations, csvs
+from df_py.volume.reward_calculator import RewardCalculator, get_df_week_number
 
 
 @enforce_types
-def _get_nft_addrs(nftvols_USD: Dict[int, Dict[str, str]]) -> List[str]:
-    """
-    @arguments
-      nftvols_USD -- dict of [chainID][nft_addr] : vol_USD_float
-
-    @return
-      nft_addrs -- list of unique nft addrs. Order is consistent.
-    """
-    nft_addrs = set()
-    for chainID in nftvols_USD:
-        for nft_addr in nftvols_USD[chainID]:
-            nft_addrs.add(nft_addr)
-    return sorted(nft_addrs)
-
-
-@enforce_types
-def _get_lp_addrs(stakes: Dict[int, Dict[str, Dict[str, float]]]) -> List[str]:
-    """
-    @arguments
-      stakes - dict of [chainID][nft_addr][LP_addr] : veOCEAN_float
-
-    @return
-      LP_addrs -- list of unique LP addrs. Order is consistent.
-    """
-    LP_addrs = set()
-    for chainID in stakes:
-        for nft_addr in stakes[chainID]:
-            for LP_addr in stakes[chainID][nft_addr]:
-                LP_addrs.add(LP_addr)
-    return sorted(LP_addrs)
-
-
-@enforce_types
-def flatten_rewards(rewards: Dict[int, Dict[str, float]]) -> Dict[str, float]:
-    """
-    @arguments
-      rewards -- dict of [chainID][LP_addr] : reward_float
-
-    @return
-      flat_rewards -- dict of [LP_addr] : reward_float
-    """
-    flat_rewards = {}
-    for chainID in rewards:
-        for LP_addr in rewards[chainID]:
-            if LP_addr not in flat_rewards:
-                flat_rewards[LP_addr] = 0.0
-            flat_rewards[LP_addr] += rewards[chainID][LP_addr]
-    return flat_rewards
-
-
-def merge_rewards(*reward_dicts):
-    merged_dict = {}
-
-    for reward_dict in reward_dicts:
-        for key, value in reward_dict.items():
-            merged_dict[key] = merged_dict.get(key, 0) + value
-
-    return merged_dict
-
-
-def calc_rewards_volume(
-    CSV_DIR, START_DATE, TOT_OCEAN, DO_PUBREWARDS=DO_PUBREWARDS, DO_RANK=DO_RANK
+def calc_volume_rewards_from_csvs(
```

Comment:
This function in line with what we discussed, great.

I just realized: it makes sense to make it csv-in _and_ csv_out.

If you look at dftool_module.py, right after this function is called, it calls:
```python
        csvs.save_volume_rewards_csv(rewperlp, csv_dir)
        csvs.save_volume_rewardsinfo_csv(rewinfo, csv_dir)
```

To me, it makes sense to put those into the function too.

----

And then: make this function call a non-csv version of the function: `calc_volume_rewards()` which is a super-thin wrapper of the OO implementation. I'd put `calc_volume_rewards()` in this module as well (calc_rewards.py).

Then the functions' roles and their naming is obvious:
- calc_volume_rewards_from_csvs
- calc_volume_rewards

Reply:
It's great to see these two functions side-by-side.

Now that I see them, they have some duplicate logic on handling defaults. DRY violation. And it's not consistent, only some logic is duplicated, not all.

I recommend: make the csvs function just a pass-through to the non-csvs function. All that csvs should do is:
- load csvs
- call non-csv function
- save csvs. 

Then have the non-csvs function handle the defaults. Basically, all the functionality that was in the csvs function should be in the non-csvs function:
```python
    prev_week = 0
    if start_date is None:
        cur_week = get_df_week_number(datetime.now())
        prev_week = cur_week - 1
    else:
        prev_week = get_df_week_number(start_date)

    if tot_ocean is None:
        tot_ocean = 0.0

    if do_pubrewards is None:
        do_pubrewards = DO_PUBREWARDS

    if do_rank is None:
        do_rank = DO_RANK
```